_aggregations/metric/scripted-metric.md

```diff
@@ -9,44 +9,220 @@ redirect_from:
+
+The following request contains all described scripts:
+
+```json
+GET transactions/_search
+{
+  "size": 0, 
+  "aggs": {
+    "total_profit": {
+      "scripted_metric": {
+        "init_script": "state.transactions = []",
+        "map_script": "state.transactions.add(doc['type'].value == 'sale' ? doc['amount'].value : -1 * doc['amount'].value)",
+        "combine_script": "double shardProfit = 0; for (t in state.transactions) { shardProfit += t; } return shardProfit;",
+        "reduce_script": "double totalProfit = 0; for (p in states) { if (p != null) { totalProfit += p; }} return totalProfit;"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+
+The response returns the `total_profit`:
+
+```json
+{
+  ...
+  "hits": {
+    "total": {
+      "value": 4,
+      "relation": "eq"
+    },
+    "max_score": null,
+    "hits": []
+  },
   "aggregations": {
-    "responses.counts": {
+    "total_profit": {
+      "value": 170
+    }
+  }
+}
+```
+
+### Categorizing HTTP response codes
+
+The following example demonstrates a more advanced use of the `scripted_metric` aggregation to return multiple values within a single aggregation. The dataset consists of web server log entries, each containing an HTTP response code. The goal is to classify the responses into three categories: successful responses (2xx status codes), client or server errors (4xx or 5xx status codes), and other responses (1xx or 3xx status codes). This classification is implemented by maintaining counters within a map-based aggregation state. 
+
+Create a sample index:
+
+```json
+PUT logs
+{
+  "mappings": {
+    "properties": {
+      "response": { "type": "keyword" }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Add sample documents with a variety of response codes:
+
+```json
+PUT logs/_bulk?refresh=true
+{ "index": {} }
+{ "response": "200" }
+{ "index": {} }
+{ "response": "201" }
+{ "index": {} }
+{ "response": "404" }
+{ "index": {} }
+{ "response": "500" }
+{ "index": {} }
+{ "response": "304" }
+```
+{% include copy-curl.html %}
+
+The `state` (on each shard) is a map with three counters: `error`, `success`, and `other`. 
+
+To run a scripted metric aggregation which counts the categories, use the following scripts:
+
+- The `init_script` initializes counters for `error`, `success`, and `other` to `0`. 
+- The `map_script` examines each document’s response code and increments the relevant counter based on the response code. 
+- The `combine_script` returns the `state.responses` map for that shard.
+- The `reduce_script` takes the array of maps (`states`) from all shards and merges them. Thus, it creates a new combined map and adds the `error`, `success`, and `other` counts from each shard’s map. This combined map is returned as the final result.
+
```

Comment:
Line 187: Please confirm that my change to the first sentence retains the intended meaning.